test/promise-dialog_test.html

```diff
@@ -85,9 +85,10 @@
           });
           it('should resolve with value', function(done) {
             let expected = 42;
-            resolveButton.setAttribute('promise-value', expected);
+            resolveButton['promise-value'] = expected;
+            resolveButton.setAttribute('promise-value', "this line does nothing");
```

Comment:
I think it would be worth while to have a test case that checks for attributes being set